### DISCUSSION TITLE
Need help undetstanding HTMX

### DISCUSSION THREAD

**Discussion by kartikynwa** [bigskysoftware/htmx#2228] [General]
Hi. I mostly have experience in Python, Rust and Go (backend). Lately I started learning Javascript and React so I am new to this realm. I came across HTMX through a link aggregator and was intrigued. HTMX looks really interesting. But I am having difficulty understanding what HTMX is fundamentally capable of so was looking for pointers regarding that.

When I see an example like:

```html
  <button hx-post="/clicked" hx-swap="outerHTML">
    Click Me
  </button>
```

It makes me think I can use a combination of HTMX and a templating language to serve interactive web pages straight from the backend.

But what I was wondering was, is it possible and practical to use HTMX to create a frontend without the backend? For example, let's say I want to create a frontend for Mastodon utilising its API. Maybe I want to make this endpoint instance-agnostic too. Is HTMX a feasible option in such cases?

Thanks for taking the time to read this.

**Comment by nickchomey** [comment]
htmx is fundamentally about html/hypermedia. So long as you receive HTML from somewhere - be it your origin server, a 3rd party api, etc... - then you can use it. 

Check out the docs, the essays and the hypermedia.systems book.
